sdk/keyvault/keyvault-admin/src/settingsClientModels.ts

```diff
@@ -36,50 +36,40 @@ export interface GetSettingOptions extends OperationOptions {}
 export interface ListSettingsOptions extends OperationOptions {}
 
 /**
- * Common attributes shared by Key Vault settings of all kinds.
+ * A Key Vault setting.
  */
-export interface KeyVaultSettingCommon {
+export interface KeyVaultSetting {
   /**
-   * The name of the setting.
+   * The kind of the setting.
    */
-  name: string;
-}
+  kind?: string;
 
-/**
- * A Key Vault setting for which the type of the value is unknown.
- */
-export interface UnknownKeyVaultSetting extends KeyVaultSettingCommon {
   /**
-   * The value of the setting.
+   * The name of the setting.
    */
-  value: unknown;
+  name: string;
 
   /**
-   * The kind of the setting.
+   * The value of a setting. The type of the value depends on the value of the kind property.
    */
-  kind?: never;
+  value: unknown;
 }
 
 /**
- * A Key Vault setting with a boolean value.
+ * A Key Vault setting of boolean type. To check if a given KeyVaultSetting is a boolean, use {@link isBooleanSetting}.
  */
-export interface BooleanKeyVaultSetting extends KeyVaultSettingCommon {
+export interface BooleanKeyVaultSetting extends KeyVaultSetting {
   /**
-   * The value of the setting.
+   * The kind of the setting.
    */
-  value: boolean;
+  kind: "boolean";
 
   /**
-   * The kind of the setting.
+   * The value of the setting as a boolean.
    */
-  kind: "boolean";
+  value: boolean;
```

Comment:
A question we're asking on the Python team right now is whether the `value` property of a KeyVaultSetting should be deserialized into the built-in type at all. This pattern works for now, and I think is convenient for booleans, but the concern is from the possibility of a new setting type being added within the same service API version.

For example, if a new setting with an integer type were added, that looks like it would be returned by `getSetting` as a general KeyVaultSetting with a string `value` -- say, `"1"`. If a later library release added support for an IntegerKeyVaultSetting that has an integer type for the `value` property, that same setting would now get returned as an IntegerKeyVaultSetting with a value of `1` (as an integer). That would be a breaking change, even if we document that it could happen.

Reply:
Yeah, that's a valid concern, but I feel that it doesn't apply for JavaScript with this new API. `KeyVaultSetting` as returned by `getSetting` has a `value` of type `unknown`, which means we make zero guarantees about the underlying type of the value (it could be a string, integer, boolean, object, anything). This means that changing the behavior to serialize a new setting type would not be a breaking change. If customers want to use the `value`, they have two safe options:

- Use the `isBooleanSetting` type guard to verify the setting is a boolean (or other type guards as they are added).
- Stringify the unknown `value` by calling `String(value)`. This is should not break people if we were to change the behavior for settings of other types and deserialize a setting value of a new type.